File: Sleep_Sort/Sleep_Sort.scala
object SleepSort {
  def sort(nums:Seq[Int])=nums foreach {n =>
    scala.concurrent.ops.spawn{
      Thread.sleep(500*n)
      print(n+" ")
    }
  }
 
  def main(args:Array[String])={
    sort(args map (_.toInt))
  }
}

/*
Output:
> scala SleepSort 1 9 8 7 6 5 3 4 5 2 0
0 1 2 3 4 5 5 6 7 8 9
*/

File: Sleep_Sort/Sleep_Sort.java
/**
 * Created by jude on 21/10/16.
 */
public class sleepSort {
    private static int[] elements = {3, 6, 3, 9, 10, 8, 2};

    public static void main(String[] args)
    {
        sleepSortMethod(elements);
    }

    private static void sleepSortMethod(int[] elements) {
        for(int element: elements)
        {
            Thread thread = new Thread(new sleepSortThread(element));
            thread.start();
        }
    }

    private static class sleepSortThread implements Runnable {
        int element;
        public sleepSortThread(int element) {
            this.element = element;
        }

        @Override
        public void run() {
            try
            {
                Thread.sleep(element);
                System.out.println(element);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }
}

/* Output
        2
        3
        3
        6
        8
        9
        10
*/
